File: 백준/Silver/1912. 연속합/연속합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int val = Integer.parseInt(st.nextToken());
        int globalMax = val;
        int localMax = val;
        for (int i = 1; i < n; i++) {
            val = Integer.parseInt(st.nextToken());
            localMax = Math.max(localMax + val, val);
            globalMax = Math.max(globalMax, localMax);
        }
        System.out.println(globalMax);
        br.close();
    }
}

File: 백준/Silver/1912. 연속합/연속합.py
import sys

n, *lst = map(int, sys.stdin.read().split())
local_max = global_max = -1234567890
for i in lst:
    local_max = max(i, local_max + i)
    global_max = max(global_max, local_max)
sys.stdout.write(str(global_max))    




